acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,53 @@
+require 'pxp-agent/test_helper.rb'
+require 'puppet/acceptance/environment_utils'
+
+test_name 'C93063 - pxp-module-puppet run expect puppet failure' do
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type = File.basename(__FILE__, '.*')
+  tmp_environment   = mk_tmp_environment(app_type)
+
+  step 'On master, create a new environment that will result in Failed' do
+    site_manifest = File.join(environmentpath, tmp_environment, 'manifests', 'site.pp')
+    create_remote_file(master, site_manifest, <<-SITEPP)
+node default {
+  fail('intended failure')
+}
+SITEPP
+  end
+
+  step 'Ensure each agent host has pxp-agent running and associated' do
+    agents.each do |agent|
+      on agent, puppet('resource service pxp-agent ensure=stopped')
+      create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_puppet_certs(master, agent).to_s)
+      on agent, puppet('resource service pxp-agent ensure=running')
+      assert(is_associated?(master, "pcp://#{agent}/agent"),
+             "Agent #{agent} with PCP identity pcp://#{agent}/agent should be associated with pcp-broker")
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents" do
+    target_identities = []
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'), "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal('failed', action_result['status'], "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: '#{action_result["status"]}'")
```

Comment:
Possibly an assertion on the error message "intended failure" could be done here; to ensure that we got our planned failure instead of some other failure